code/src/OSFramework/Grid/AbstractDataSource.ts

```diff
@@ -247,7 +247,9 @@ namespace OSFramework.Grid {
             this._metadata = dataJson.metadata;
 
             this._isSingleEntity =
-                Object.keys(dataJson[0] || dataJson.data[0] || {}).length <= 1;
+                Object.keys(
+                    this._metadata || dataJson[0] || dataJson.data[0] || {}
```

Comment:
@glundgren93 you should remove this part `dataJson.data[0]` because:
1. it's not doing anything :)
2. if the developer does not use our action and passes an {}, it will throw an exception...

Reply:
fixed